components/my-account/Lists/CreateList/CreateList.tsx

```diff
@@ -0,0 +1,262 @@
+import React, { FormEvent, useState } from 'react'
+
+import ArrowBackIosIcon from '@mui/icons-material/ArrowBackIos'
+import { Button, useMediaQuery, useTheme, IconButton, Box, Typography } from '@mui/material'
+import { useRouter } from 'next/router'
+import { useTranslation } from 'next-i18next'
+
+import ProductSearch from '@/components/b2b/B2BProductSearch/B2BProductSearch'
+import { KiboTextBox } from '@/components/common'
+import styles from '@/components/my-account/Lists/CreateList/CreateList.style'
+import ListItem from '@/components/my-account/Lists/ListItem/ListItem'
+import { useAuthContext } from '@/context'
+import { useCreateWishlist, useGetWishlist } from '@/hooks'
+
+import { CrProductPrice, Product } from '@/lib/gql/types'
+
+export interface CreateListProps {
+  onCreateFormToggle: (param: boolean) => void
+}
+
+interface CreateListState {
+  name: string
+  items: [
+    {
+      product: {
+        productCode: string
+      }
+      quantity: number
+    }
+  ]
+}
+
+const CreateList = (props: CreateListProps) => {
+  const { onCreateFormToggle } = props
+  const [listState, setListState] = useState<CreateListState>({
+    name: '',
+    items: [{ product: { productCode: '' }, quantity: 0 }],
+  })
+  const [productList, setProductList] = useState<Product[]>([])
+
+  const theme = useTheme()
+  const mdScreen = useMediaQuery<boolean>(theme.breakpoints.up('md'))
+  const router = useRouter()
+  const { t } = useTranslation('common')
+  const { user } = useAuthContext()
+  const { createWishlist } = useCreateWishlist()
+  const { refetch } = useGetWishlist()
+
+  const handleSubmit = async (e: FormEvent) => {
+    e.preventDefault()
+    const finalList = listState.items
+    finalList.shift()
+    debugger
+    await createWishlist
+      .mutateAsync({
+        customerAccountId: user?.id,
+        name: listState.name,
+        items: finalList,
+      })
+      .catch((e) => {
+        console.log(e)
+        alert('error occured')
+      })
+    await refetch()
+    setListState({ name: '', items: [{ product: { productCode: '' }, quantity: 0 }] })
+    onCreateFormToggle(false)
+  }
```

Comment:
are we not invalidating key? why we need to refetch?

Reply:
done